macros/generic_tests/test_not_null.sql

```diff
@@ -11,15 +11,28 @@
 {# Only execute test when field exists. Otherwise execute a dummy test that always succeeds. #}
 {% if column_name in column_names %}
     {# Query the records that fail the test. #}
-    select {{ column_name }}
-    from {{ model }}
-    where {{ column_name }} is null or len({{ column_name }}) = 0
+    {%- if target.type == 'snowflake' -%}
+        select {{ column_name }}
+        from {{ model }}
+        where len({{ column_name }}) is null or len({{ column_name }}) = 0
```

Comment:
This can be simplified. In the first clause `{{ column_name }} is null` would achieve the same.

Reply:
Same comment as above